### PR TITLE
e2e: add delete dc, auto failover, upgrade test cases

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve?

- add delete dc case
- add auto failover for DM case
- add upgrade DM case

part of #2868

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: tests/actions.go [hunk 1]
```diff
@@ -3601,6 +3628,22 @@ func (oa *OperatorActions) WaitForDmClusterReady(dc *v1alpha1.DMCluster, timeout
 	})
 }
 
+func (oa *OperatorActions) WaitForDmClusterDeleted(ns, dcName string, timeout, pollInterval time.Duration) error {
```
Comment (july2993) [inline tests/actions.go:3631]: do not log, but return the reason about why check failed when timeout?

**Review by july2993 — APPROVED** [review]

File: tests/actions.go (hunk 1 above)
Comment (csuzhangxc) [reply tests/actions.go:3633]: :+1: updated in https://github.com/pingcap/tidb-operator/pull/3982/commits/3c184d2f443b957cd5a873ef74416730c1928fac.

**Review by july2993 — APPROVED** [review]

**Pull request merged**